src/kubectl.ts

```diff
@@ -81,6 +85,21 @@ async function invoke(context : Context, command : string, handler? : ShellHandl
     await kubectlInternal(context, command, handler || kubectlDone(context));
 }
 
+async function invokeWithProgress(context : Context, command: string, progressMessage: string, handler?: ShellHandler): Promise<{}> {
```

Comment:
The combination of `async` + callback dates back to when I was starting to transition the code away from Node.js-style callbacks to TypeScript async/await.  We shouldn't really need both - we only have it in `kubectl.invoke` because a lot of the existing code relied on the callback (specifically the wrapper that injected `kubectlDone` as the default handler) and I didn't want to have to rewrite it all.  It would be desirable for new functions to use `async`, return a `Promise`, in this case a `Promise<ShellResult>`, and not take a callback.  It does bother me though if every call site has to call `kubectlDone`... so perhaps it's best to leave this as you have it for now, and address both it and `invoke` later.  Thoughts?

Reply:
i would prefer to add two interfaces `invokeWithProgress` and `invokeAsyncWithProgress` for the corresponding `invoke` and `invokeAsync`. For old invoke code, we could choose to use `invokeWithProgress` with minor effort.